operator/pkg/component/controlplane/control_plane.go

```diff
@@ -55,34 +50,24 @@ func NewIstioOperator(installSpec *v1alpha1.IstioOperatorSpec, translator *trans
 	}
 
 	for idx, c := range installSpec.Components.IngressGateways {
-		if c.Name == istioIngressGatewayName {
-			enabled, pathExist, err := translate.IsComponentEnabledFromValue(name.IngressComponentName, installSpec.Values)
-			if err == nil && pathExist {
-				if c.Enabled == nil {
-					c.Enabled = &v1alpha1.BoolValueForPB{}
-				}
-				c.Enabled.Value = enabled
+		enabled, pathExist, err := translate.IsComponentEnabledFromValue(name.IngressComponentName, installSpec.Values)
```

Comment:
i assume that this is intended to be a generalization that would work also for user defined gateways. if so, we need to match the user gateway name so that the correct gateway is selected. 
i recommend starting with a unit test that has the default istio-ingressgateway and a user gateway to check if your code correctly handles all combinations of 00, 01, 10 and 11 enablement of the two.

Reply:
@richardwxn

Reply:
added a test case which has multiple custom gateways. 
if k.Name == rn && (k.Namespace == ns || k.Namespace == "")  --- this is use ingress gateway custom name to check whether it is enabled or not.